tests/jballerina-unit-test/src/test/resources/test-src/object/object_self_final_negative.bal

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+class Class {
+    int i;
+
+    function init(int i) {
+        self.i = i;
+    }
+
+    function fn() {
+        self = new (3);
```

Comment:
Was this supported previously? I hope no one is misusing this bug. :)

Reply:
Yes, it was. But it didn't work even as one would have expected (`self` isn't updated).